pkg/audit/audit.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package audit
+
+import (
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/requestutil"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type AuditConfig struct {
+	Label []string
+}
+type TypeMatcher interface {
+	MatchType([]string) bool
+}
+
+type SinkTypeMatcher struct {
+	Type string
+}
+
+func (m *SinkTypeMatcher) MatchType(types []string) bool {
+	for _, item := range types {
+		if m.Type == item {
+			return true
+		}
+	}
+	return false
+}
+
+type Sink interface {
+	ProcessRequest(event requestutil.RequestInfo) bool
+	TypeMatcher
+}
+
+type MainLogSink struct {
+	TypeMatcher
+}
+
+func (l *MainLogSink) ProcessRequest(event requestutil.RequestInfo) bool {
+	fields, _ := convertRequestEventToZapFields(event)
+	log.Info("Audit Log", fields...)
+	return true
+}
+
+func convertRequestEventToZapFields(event requestutil.RequestInfo) ([]zapcore.Field, error) {
```

Comment:
Can we add a String method instead of append here?

Reply:
fix